objects/src/notes/output.rs

```diff
@@ -0,0 +1,99 @@
+use miden_crypto::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{NoteConsumable, NoteEnvelope, NoteId, NoteMetadata, NoteVault};
+use crate::Digest;
+
+const PUBLIC: u8 = 0;
+const PRIVATE: u8 = 1;
+
+/// The types of note outputs supported by the transaction kernel.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum NoteOutput {
+    Public(NoteConsumable),
+    Private(NoteEnvelope),
+}
```

Comment:
Similarly (or alternatively) to the other note type validation comment, it might be a good idea to validate on creation that `NoteOutput::Private(envelope)` cannot contain an envelope with `NoteType::Onchain` and the reverse for `NoteOutput::Public`

Reply:
Added validation